### ISSUE TITLE
Switch anstyle-roff to anstyle-parse

### ISSUE THREAD

**Issue by epage** [rust-cli/anstyle#63]
With #4, we can now use our own parser.  We might need to reuse some code from `anstyle-stream` to do so.

**Comment by Calder-Ty** [comment]
Hey I saw this, and saw the work you did forking vte. With the transfer of the
Roff style code were you thinking of a general purpose `bytes` or `str` to `anstyle` or just
implement one for the `roff` crate for now?

**Comment by epage** [comment]
Yes, I think it could make sense to make the conversion generic since roff will likely need everything that an anstyle parser would need.

`anstyle-stream` has a subset of bytes to anstyle used, just enough to get wincon working.  That could be generalized.  I lean towards that living in `anstyle_parse` or `anstyle_stream`.  No preference yet on which of those it should be in.

This also isn't the most critical of things; I just didn't want to lose track of this.
